### PR TITLE
CO-948: Add integration test for isOpen

### PR DESCRIPTION
**Pull request by tsoliangwu0130**

### CODE REVIEW

File: src/test/integration/integrationtests.py [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+                        check_ssl, \
+                        get_buildings_with_services, \
+                        id_request, \
+                        not_found_request, \
+                        query_request, \
+                        response_time, \
+                        results_with_links, \
+                        unauth_request
+from configuration_load import get_access_token, get_single_resource_id, get_url
+
```
Comment (jaredkosanovic) [inline src/test/integration/integrationtests.py:16]: extra blank line?

File: src/test/integration/integrationtests.py [hunk 2]
```diff
@@ -221,11 +262,12 @@ def test_ssl_v3(self):
             self.skipTest('SSLv3 support not available')
         self.assertFalse(check_ssl(ssl.PROTOCOL_SSLv3, locations_url, access_token))
 
+
```
Comment (jaredkosanovic) [inline src/test/integration/integrationtests.py:256]: extra blank line?

File: src/test/integration/integrationtests.py [hunk 3]
```diff
@@ -145,14 +156,44 @@ def test_parking(self):
...
+                open_hours = open_resource['attributes']['openHours']
+                self.assertIsNotNone(open_hours)
+
+                # If there are not only one open hours, current time should be in one of these open hours intervals
+                if len(open_hours[weekday]) > 1:
+                    is_open = False
+                    for open_hour in open_hours[weekday]:
+                        if open_hour['start'][:-1] < now < open_hour['end'][:-1]:
+                            is_open = True
+                            self.assertTrue(is_open)
```
Comment (jaredkosanovic) [inline src/test/integration/integrationtests.py]: What if none of the time slots contain the current time? This test would never fail, it just wouldn't run, yes? If none of the time slots surround the current time, that should be a fail.

File: src/test/integration/integrationtests.py [hunk 4]
```diff
@@ -162,6 +163,37 @@ def test_parking(self):
...
+            for open_resource in all_open_resources['data']:
+                # Test that only open resources are returned when they should be and each open resource has a related open hours
+                open_hours = open_resource['attributes']['openHours']
+                self.assertIsNotNone(open_hours)
+
+                # If there are not only one open hours, current time should be in one of these open hours intervals
+                if len(open_hours[weekday]) > 1:
+                    is_open = False
+                    for open_hour in open_hours[weekday]:
+                        if open_hour['start'][:-1] < now < open_hour['end'][:-1]:
```
Comment (magical) [inline src/test/integration/integrationtests.py]: is the `[:-1]` to strip the trailing Z? can we do that explicitly instead? `open_hour['start'].rstrip('Z')`. or just add a trailing Z to `now`.

File: src/test/integration/integrationtests.py (hunk 3 above)
Comment (magical) [reply src/test/integration/integrationtests.py]: yes. i think this line needs to be dedented by a couple blocks. it seems like what we really want to say is `self.assertTrue( any(open_hour['start'] <= now+'Z' <= open_hour['end'] for open_hour in open_hours[weekday]) )`

File: src/test/integration/integrationtests.py (hunk 2 above)
Comment (tsoliangwu0130) [reply src/test/integration/integrationtests.py:256]: hmm, in pep8/flake8, we generally use two blank lines between two def sections. See error code E305 here: https://github.com/PyCQA/pycodestyle/blob/master/docs/intro.rst#error-codes

File: src/test/integration/integrationtests.py (hunk 3 above)
Comment (tsoliangwu0130) [reply src/test/integration/integrationtests.py]: @jaredkosanovic good catch. Will fix it.

File: src/test/integration/integrationtests.py (hunk 4 above)
Comment (tsoliangwu0130) [reply src/test/integration/integrationtests.py]: Agree. Will fix.

File: src/test/integration/integrationtests.py [hunk 5]
```diff
@@ -183,13 +183,9 @@ def test_resource(resource_url):
 
                 # If there are not only one open hours, current time should be in one of these open hours intervals
                 if len(open_hours[weekday]) > 1:
```
Comment (magical) [inline src/test/integration/integrationtests.py]: I think this if statement is unnecessary now.

Comment (tsoliangwu0130) [reply src/test/integration/integrationtests.py]: ha you are right!

**Review by jaredkosanovic — APPROVED** [review]

**Pull request merged**